packages/mwp-app-render/src/components/uxcapture/UXCaptureStartView.jsx

```diff
@@ -8,56 +8,70 @@ type Props = {|
 	secondaryContentDisplayed?: Array<string>,
 |};
 
-export default ({
-	destinationVerified,
-	primaryContentDisplayed,
-	primaryActionAvailable,
-	secondaryContentDisplayed,
-}: Props) => {
-	const zones = [];
-
-	if (destinationVerified && destinationVerified.length > 0) {
-		zones.push({
+const makeZones = (props: Props) =>
+	[
+		{
 			name: 'ux-destination-verified',
-			marks: destinationVerified,
-		});
-	}
-
-	if (primaryContentDisplayed && primaryContentDisplayed.length > 0) {
-		zones.push({
+			marks: props.destinationVerified,
+		},
+		{
 			name: 'ux-primary-content-displayed',
-			marks: primaryContentDisplayed,
-		});
-	}
-
-	if (primaryActionAvailable && primaryActionAvailable.length > 0) {
-		zones.push({
+			marks: props.primaryContentDisplayed,
+		},
+		{
 			name: 'ux-primary-action-available',
-			marks: primaryActionAvailable,
-		});
-	}
-
-	if (secondaryContentDisplayed && secondaryContentDisplayed.length > 0) {
-		zones.push({
+			marks: props.primaryActionAvailable,
+		},
+		{
 			name: 'ux-secondary-content-displayed',
-			marks: secondaryContentDisplayed,
-		});
-	}
+			marks: props.secondaryContentDisplayed,
+		},
+	].filter(({ marks }) => marks && marks.length > 0);
 
-	// Don't add UX.expect() to page is no zones specified
-	if (!zones.length) {
-		return null;
+export default class UXCaptureStartView extends React.Component<Props> {
+	zones = makeZones(this.props);
+	componentDidUpdate(prev: Props) {
+		// updated on client - if mark name has changed, clear old mark and trigger new
+		if (
+			prev.destinationVerified !== this.props.destinationVerified &&
```

Comment:
Should this trigger if only one of the props has changed and not all of them?

Reply:
I _think_ so, but it sort of depends on what kind of rules we think we can enforce about mark naming - I can definitely imagine different related views (e.g. tabs) re-using the zone definitions but changing only 1 or 2 expected mark names, so I think we need to look for pretty small prop changes